BridgeAppSDK/SBAProfileItem.swift

```diff
@@ -163,65 +169,103 @@ extension SBAProfileItem {
         
         switch self.itemType {
         case SBAProfileTypeIdentifier.string:
-            return newValue as? String != nil
+            return true // anything can be cast to a string
             
         case SBAProfileTypeIdentifier.number:
-            return newValue as? NSNumber != nil
+            if (newValue as? NSNumber != nil) {
+                return true
+            }
+            else if let quantity = newValue as? HKQuantity {
+                return quantity.is(compatibleWith: self.unit ?? commonDefaultUnit())
+            }
+            return false
             
         case SBAProfileTypeIdentifier.bool:
             return newValue as? NSNumber != nil
             
         case SBAProfileTypeIdentifier.date:
-            return newValue as? NSDate  != nil
+            return newValue as? NSDate != nil
             
         case SBAProfileTypeIdentifier.hkBiologicalSex:
             return newValue as? HKBiologicalSex != nil
             
         case SBAProfileTypeIdentifier.hkQuantity:
-            return newValue as? NSNumber != nil
+            guard let quantity = newValue as? HKQuantity else { return false }
+            return quantity.is(compatibleWith: self.unit ?? commonDefaultUnit())
             
         default:
-            return false
+            return true   // Any extended type isn't included in the common validation
```

Comment:
This came up b/c Eric modeled the externalId to return a number-type question result. I ended up removing that value from the profile items, but it did occur to me that the server handles strings this way so it seemed as if there was precedent.